### ISSUE TITLE
QuantifiedConstraints parse error

### ISSUE THREAD

**Issue by georgefst** [ndmitchell/hlint#866]
Hlint reports a parse error at the line:
```haskell
deriving instance (forall a. Show (t a)) => Show (E t c)
```
Despite trying everything I can to tell hlint to enable `QuantifiedConstraints`.

GHC is fine with it.

It's very possible this has the same cause as #850.

**Comment by georgefst** [comment]
Ok, I was running an older version of hlint than I realised. This appears to have been fixed.

**Comment by georgefst** [comment]
Well actually, there is something odd going on here, but maybe it's an issue with `hie`.

On version `2.2.10`, hlint is able to parse the file when I use the standalone executable, but not when used as part of `hie` (the parser reports 'Illegal class assertion').

I have checked, and my `hie` executable was built with `hlint-2.2.10`.

**Comment by ndmitchell** [comment]
I'd bring the issue over to hie - if it works standalone then I guess it's something hie is doing? Or maybe they are using the API which is doing the wrong thing?

**Comment by ndmitchell** [comment]
HIE depended on an old version, so they can fix it.
